components/director/internal/domain/scenarioassignment/engine.go

```diff
@@ -50,38 +48,6 @@ func NewEngine(labelService labelUpsertService, labelRepo labelRepository, scena
 	}
 }
 
-// EnsureScenarioAssigned ensures that the scenario is assigned to all the runtimes that are in the ASAs target_tenant_id
-func (e *engine) EnsureScenarioAssigned(ctx context.Context, in model.AutomaticScenarioAssignment) error {
```

Comment:
Maybe we can moove the other two methods as well as it does not makes much sense to encapsulate only those two in a separate "engine".

If you decided to leave it like this there are labelRepo and labelService which seems unused now.

Reply:
I'm not sure whether or not they should be moved or where exactly those methods should reside, I'll think about what will need to change and what might occur before moving them because this could add more dependency issues as the methods are used in the runtime service. For now I'll remove the unused fields and start from there.

Reply:
If we move them to the runtime service, as that's where they're primarily used, that would force the formation service to depend on the runtime service, which might become a problem. But if it's moved to the formations service it would require the runtime service to depend on the formations service, which also could prove to be a problem in the future. Out of the two options, the latter makes more sense, but requires more changes. Leaving it as is will still probably require a change in the future.

Reply:
The runtime service will depend on the formation service anyways once we adapt the runtime mutations as part of a separate task. So, maybe we can move them in the formation service if there are no major obstacles right now.

Reply:
I will start on moving them to the formations service right now in that case.